recipe/magento2.php

```diff
@@ -132,65 +154,28 @@
 });
 
+desc('Set maintenance mode if needed');
+task('magento:maintenance:enable-if-needed', function () {
+    ! get('enable_zerodowntime') || get('database_upgrade_needed') || get('config_import_needed') ?
+        invoke('magento:maintenance:enable') :
+        writeln('Config and database up to date => no maintenance mode');
+});
+
 desc('Config Import');
 task('magento:config:import', function () {
-    $configImportNeeded = false;
-
-    if(version_compare(get('magento_version'), '2.2.0', '<')) {
-        //app:config:import command does not exist in 2.0.x and 2.1.x branches
-        $configImportNeeded = false;
-    } elseif(version_compare(get('magento_version'), '2.2.4', '<')) {
-        //app:config:status command does not exist until 2.2.4, so proceed with config:import in every deploy
-        $configImportNeeded = true;
+    if (get('config_import_needed')) {
+        run('{{bin/php}} {{release_or_current_path}}/{{bin/magento}} app:config:import --no-interaction');
     } else {
-        try {
-            run('{{bin/php}} {{release_or_current_path}}/bin/magento app:config:status');
-        } catch (RunException $e) {
-            if ($e->getExitCode() == CONFIG_IMPORT_NEEDED_EXIT_CODE) {
-                $configImportNeeded = true;
-            } else {
-                throw $e;
-            }
-        }
-    }
-
-    if ($configImportNeeded) {
-        if (get('enable_zerodowntime') && !get('maintenance_mode_status_active')) {
-            invoke('magento:maintenance:enable');
-        }
-
-        run('{{bin/php}} {{release_or_current_path}}/bin/magento app:config:import --no-interaction');
-
-        if (get('enable_zerodowntime') && !get('maintenance_mode_status_active')) {
-            invoke('magento:maintenance:disable');
-        }
+        writeln('App config is up to date => import skipped');
     }
 });
 
 desc('Upgrades magento database');
 task('magento:upgrade:db', function () {
-    $databaseUpgradeNeeded = false;
-
-    try {
-        run('{{bin/php}} {{release_or_current_path}}/bin/magento setup:db:status');
-    } catch (RunException $e) {
-        if ($e->getExitCode() == DB_UPDATE_NEEDED_EXIT_CODE) {
-            $databaseUpgradeNeeded = true;
-        } else {
-            throw $e;
-        }
-    }
-
-    if ($databaseUpgradeNeeded) {
-        if (get('enable_zerodowntime') && !get('maintenance_mode_status_active')) {
-            invoke('magento:maintenance:enable');
-        }
-
-        run("{{bin/php}} {{release_or_current_path}}/bin/magento setup:upgrade --keep-generated --no-interaction");
-
-        if (get('enable_zerodowntime') && !get('maintenance_mode_status_active')) {
-            invoke('magento:maintenance:disable');
-        }
+    if (get('database_upgrade_needed')) {
+        run("{{bin/php}} {{release_or_current_path}}/{{bin/magento}} setup:upgrade --keep-generated --no-interaction");
```

Comment:
Again, see https://github.com/deployphp/deployer/pull/3460#discussion_r1091092853

Reply:
I would like to wait until https://github.com/deployphp/deployer/pull/3460 is merged to fix this

Reply:
Rebased and fixed `bin/magento` calls :heavy_check_mark: